sox.features.js

```diff
@@ -705,28 +705,34 @@
         $(document.getElementById('question-header')).prepend(getHotDiv());
       }
 
-      if (sox.location.on('/questions') || $('.question-summary').length) {
-        const proxyUrl = 'https://cors-anywhere.herokuapp.com/'; //CORS proxy
-        const hnqJSONUrl = 'https://stackexchange.com/hot-questions-for-mobile';
-        const requestUrl = proxyUrl + hnqJSONUrl;
-
-        $.get(requestUrl, results => {
-          if (sox.location.on('/questions/')) {
-            $.each(results, (i, o) => {
-              if (document.URL.indexOf(o.site + '/questions/' + o.question_id) > -1) addHotText();
-            });
-          } else {
-            $('.question-summary').each(function() {
-              const id = $(this).attr('id').split('-')[2];
-              if (results.filter(d => {
-                return d.question_id == id;
-              }).length) {
-                $(this).find('.summary h3').prepend(getHotDiv('question-list'));
-              }
-            });
-          }
-        });
+      const sitename = sox.site.currentApiParameter;
+
+      if (sox.location.on('/questions')) {
+        const postId = window.location.pathname.split('/')[2]
+        apiCall(postId, sitename);
+      } else if ($('.question-summary').length) {
+          $('.question-summary').each(function() {
+              const postID = $(this).attr('id').split('-')[2];
+              apiCall(postID, sitename, this);
+          });
       }
+        function apiCall(postID, sitename, el) {
+            sox.helpers.getFromAPI({
+                endpoint: 'posts',
+                childEndpoint: 'revisions',
+                sitename: sitename,
+                filter: '!5RC-)9_aw3mg*i)3*vUhU3Wfl',
+                ids: postID,
+                featureId: 'isQuestionHot',
+                cacheDuration: 60 * 8, // Cache for 8 hours
+            }, results => {
+                $.each(results, (i, o) => {
```

Comment:
I think it might be cleaner to use a vanilla JS loop here, as `results` is just an array and we're not using `o` anywhere. We could use either `forEach` or a normal `for` loop

Reply:
@shu8 how about `$.each(results, i =>`? Won't that work, too? No `o` will be used.

Reply:
yeah we could do that, but I think it's better to use vanilla JS where we can/isn't much harder than jquery, and the jquery $.each isn't really useful in this case

Reply:
OK, that makes sense. I have implemented what you said - now a `forEach()` loop is used!